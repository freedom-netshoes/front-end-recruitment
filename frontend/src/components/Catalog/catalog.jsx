import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addProductCart, toggleCart } from '../../actions/action'
import { WebConfig } from '../../resources/appSettings'
import CSSModules from 'react-css-modules'
import axios from 'axios'

import styles from './catalog.less'

import Product from '../Product/product'

@CSSModules(styles)
class Catalog extends Component {
  constructor (props) {
    super(props)
    this.state = {
      products: []
    }
  }

  componentDidMount () {
    this.loadItems()
  }

  loadItems = () => {
    axios.get(WebConfig.apiProductUrl).then(res => this.setState({ ...this.state, products: res.data }))
  }

  render () {
    return (
      <div className='row center-xs' styleName={this.props.cartState ? 'fixed' : ''}>
        <section id='products' className='col-xs-7'>
          <ul className='row'>
            {
              this.state.products.map(product =>
                <li {...{
                  onClick: () => {
                    this.props.dispatch(addProductCart(product))
                    this.props.dispatch(toggleCart(true))
                  },
                  className: 'col-md-4 col-sm-6 col-xs-12',
                  styleName: 'row',
                  key: product.id
                }}>
                  <Product {...{ ...product }} />
                </li>
              )
            }
          </ul>
        </section>
      </div>
    )
  }
}
const mapStateToProps = state => {
  return {
    cartState: state.cartState
  }
}
export default connect(mapStateToProps)(Catalog)
