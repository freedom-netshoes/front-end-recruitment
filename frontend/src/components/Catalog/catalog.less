.row {
    margin-top: 20px;
}

.fixed {
    overflow: hidden
}