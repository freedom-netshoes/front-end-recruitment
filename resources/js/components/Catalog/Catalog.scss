@import '../../../scss/shared';

.products {
  padding-top: 90px;

  li {
    padding-bottom: 1rem;
    padding-top   : 1rem;

    a {
      display: block;
    }
  }
}