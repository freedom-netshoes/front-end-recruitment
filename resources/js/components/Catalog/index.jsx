import React, {Component, PropTypes} from 'react';
import CSSModules from 'react-css-modules';
import Installments from 'Installments';
import Photo from 'Photo';
import Price from 'Price';
import Title from 'Title';
import styles from './Catalog.scss';

@CSSModules(styles)
class Catalog extends Component {

  static defaultProps = {

  };

  static propTypes = {

  };

  render () {
    return (
      <section {...{
        className: 'col-xs-12',
        id: 'products',
        styleName: 'products'
      }}>
        <ul className='row'>
          <li className='col-md-4 col-sm-6 col-xs-12 center-xs'>
            <a href='javascript:void(0);' title='Chuteira Nike'>
              <Photo />
              <Title />
              <Price />
              <Installments />
            </a>
          </li>
          <li className='col-md-4 col-sm-6 col-xs-12 center-xs'>
            <a href='javascript:void(0);' title='Chuteira Nike'>
              <Photo />
              <Title />
              <Price />
              <Installments />
            </a>
          </li>
          <li className='col-md-4 col-sm-6 col-xs-12 center-xs'>
            <a href='javascript:void(0);' title='Chuteira Nike'>
              <Photo />
              <Title />
              <Price />
              <Installments />
            </a>
          </li>
          <li className='col-md-4 col-sm-6 col-xs-12 center-xs'>
            <a href='javascript:void(0);' title='Chuteira Nike'>
              <Photo />
              <Title />
              <Price />
              <Installments />
            </a>
          </li>
          <li className='col-md-4 col-sm-6 col-xs-12 center-xs'>
            <a href='javascript:void(0);' title='Chuteira Nike'>
              <Photo />
              <Title />
              <Price />
              <Installments />
            </a>
          </li>
          <li className='col-md-4 col-sm-6 col-xs-12 center-xs'>
            <a href='javascript:void(0);' title='Chuteira Nike'>
              <Photo />
              <Title />
              <Price />
              <Installments />
            </a>
          </li>
          <li className='col-md-4 col-sm-6 col-xs-12 center-xs'>
            <a href='javascript:void(0);' title='Chuteira Nike'>
              <Photo />
              <Title />
              <Price />
              <Installments />
            </a>
          </li>
        </ul>
      </section>
    );
  }
}

export default Catalog;
