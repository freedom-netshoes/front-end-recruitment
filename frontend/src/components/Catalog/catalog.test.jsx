/* global describe, it, expect */
import React from 'react'
import Catalog from './catalog'
import { shallow } from 'enzyme'
import getStore from '../../resources/store'

describe('Testing <Catalog /> component', () => {
  it('renders the Catalog component', () => {
    const component = shallow(<Catalog store={getStore()} />)
    expect(component).toHaveLength(1)
  })

  it('testing the Catalog snapshot', () => {
    const component = shallow(<Catalog store={getStore()} />)
    expect(component).toMatchSnapshot()
  })
})
