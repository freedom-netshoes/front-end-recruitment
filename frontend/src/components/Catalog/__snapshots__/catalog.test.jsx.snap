// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Catalog /> component testing the Catalog snapshot 1`] = `
<Catalog
  cartState={false}
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
